packages/in-memory-live-query-store/src/InMemoryLiveQueryStore.ts

```diff
@@ -13,32 +25,93 @@ type StoreRecord = {
   executeOperation: () => Promise<ExecutionResult>;
 };
 
+const isIDScalarType = (type: GraphQLOutputType): type is GraphQLScalarType => {
+  if (isNonNullType(type)) {
+    return isScalarType(type.ofType);
+  }
+  return false;
+};
+
+const attachIdCollectorToSchema = (schema: GraphQLSchema): GraphQLSchema => {
+  return wrapSchema(schema, [
+    new TransformObjectFields((typeName, fieldName, fieldConfig) => {
+      if (fieldName === "id" && isIDScalarType(fieldConfig.type)) {
+        let resolve = fieldConfig.resolve;
+        fieldConfig.resolve = (async (src, args, context, info) => {
+          const id = await resolve!(src, args, context, info);
+          context?.__gatherId?.(typeName, id);
+          return id;
+        }) as GraphQLFieldResolver<any, any, any>;
+      }
+      return fieldConfig;
+    }),
+  ]);
+};
+
 export class InMemoryLiveQueryStore implements LiveQueryStore {
   private _store = new Map<DocumentNode, StoreRecord>();
+  // cache that stores all patched schema objects
+  private _cache = new Map<GraphQLSchema, GraphQLSchema>();
 
   register({
+    schema: inputSchema,
     operationDocument,
+    rootValue,
+    contextValue,
+    operationVariables,
     operationName,
-    executeOperation,
     publishUpdate,
   }: LiveQueryStoreRegisterParameter): UnsubscribeHandler {
     const [liveQuery] = extractLiveQueries(operationDocument);
     if (!liveQuery) {
       throw new Error("Cannot register live query for the given document.");
     }
 
-    const identifier = extractLiveQueryRootFieldCoordinates(
+    const rootFieldIdentifier = extractLiveQueryRootFieldCoordinates(
       operationDocument,
       operationName
     );
+
+    let schema = this._cache.get(inputSchema);
+    if (!schema) {
+      schema = attachIdCollectorToSchema(inputSchema);
+      this._cache.set(inputSchema, schema);
+    }
+
+    // keep track that current execution is the latest in order to prevent race-conditions :)
+    let executionCounter = 0;
+
     const record = {
       publishUpdate,
-      identifier,
-      executeOperation,
+      identifier: [...rootFieldIdentifier],
+      executeOperation: () => {
+        executionCounter = executionCounter + 1;
+        const counter = executionCounter;
+        const newIds: string[] = [];
+        return graphql({
+          schema,
+          source: print(operationDocument),
+          operationName,
+          rootValue,
+          contextValue: {
+            // @ts-ignore
+            ...contextValue,
+            __gatherId: (typeName, id) => newIds.push(`${typeName}:${id}`),
+          },
```

Comment:
Need to handle the case where `contextValue` is not an object. Suggestions welcome. The easiest solution would be to disable resource identifier gathering.